spec/hyacinth/adapters/preservation_adapter/fedora3_spec.rb

```diff
@@ -0,0 +1,283 @@
+      let(:digital_object_title) { "Assigned Label" }
+      before do
+        FactoryBot.create(:export_rule) # creates descMetadata
+        expect(connection).to receive(:datastream_profile).with(object_pid, described_class::HYACINTH_CORE_DATASTREAM_NAME, nil, nil).and_return({})
+        # test rels-ext elsewhere
+        allow(connection).to receive(:add_relationship)
+      end
+      it "persists templated field exports to datastreams" do
+        hyacinth_object.dynamic_field_data['name'] = [{ 'role' => "Farmer" }]
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+        actual_xml = rubydora_object.datastreams["descMetadata"].content
+        actual_xml.sub!(/^<\?.+\?>/, '') # remove PI
+        actual_xml.gsub!('mods:', '') # remove ns
+        actual_xml.gsub!(/\s/, '') # remove whitespace
+        expect(actual_xml).to eql("<mods><name>Farmer</name></mods>")
+      end
+      it "makes a core object property assignments" do
+        hyacinth_object.dynamic_field_data['title'] = [{ 'title_sort_portion' => digital_object_title }]
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+        expect(rubydora_object.label).to eql(digital_object_title)
+        expect(rubydora_object.state).to eql('A')
+      end
+      it "assigns a state property to inactive if withdrawn" do
+        hyacinth_object.state = Hyacinth::DigitalObject::State::WITHDRAWN
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+        expect(rubydora_object.state).to eql('I')
+      end
+    end
+    context "basic rels-ext properties" do
+      let(:dsids) { ['structMetadata'] }
+      let(:digital_object_title) { "Assigned Label" }
+      let(:model_property) do
+        {
+          predicate: "info:fedora/fedora-system:def/model#hasModel",
+          object: "info:fedora/ldpd:ContentAggregator",
+          pid: object_pid
+        }
+      end
+      let(:rdftype_property) do
+        {
+          predicate: "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
+          object: "http://purl.oclc.org/NET/CUL/Aggregator",
+          pid: object_pid
+        }
+      end
+      let(:restriction_property) do
+        {
+          predicate: "http://www.loc.gov/premis/rdf/v1#hasRestriction",
+          object: "onsite restriction",
+          pid: object_pid,
+          isLiteral: true
+        }
+      end
+      before do
+        expect(connection).to receive(:datastream_profile).with(object_pid, described_class::HYACINTH_CORE_DATASTREAM_NAME, nil, nil).and_return({})
+        expect(connection).to receive(:add_relationship).with(model_property)
+        expect(connection).to receive(:add_relationship).with(rdftype_property)
+        expect(connection).to receive(:add_relationship).with(restriction_property)
+      end
+      it "persists model properties" do
+        hyacinth_object.dynamic_field_data['restrictions'] = { 'restricted_onsite' => true }
```

Comment:
See earlier comment about 'restrictions' not belonging inside dynamic_field_data.